bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -114,24 +114,45 @@ private void resetStatus() {
 		status = null;
 	}
 
+	@Override
+	public boolean isWholeWordSearchAvailable(String findString) {
+		if (!isWord(findString)) { // $NON-NLS-1$
+			return false;
+		}
+
+		return !isRegExSearchAvailableAndActive();
```

Comment:
Shorter and easier to read:
```suggestion
		return isWord(findString) && !isRegExSearchAvailableAndActive();
```

Reply:
Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/a0b224321663e63aa99f60da4a2527f993d68f7d